src/services/Store/Contact/useContacts.tsx

```diff
@@ -65,10 +67,24 @@ function useContacts() {
     dispatch(createAContact({ ...item, uuid }));
   };
 
+  const createContacts = (items: Contact[]) => {
+    dispatch(
+      createManyContacts(
+        items.map((item) => 
+          ({ ...item, uuid: generateDeterministicAddressUUID(item.network, item.address) })
+        )
+      )
+    );
+  };
```

Comment:
Caller should be obnoxious of the uuid format.
Use action creator for setting the uuid.

```
type createContact(items: Contact[], generateUUID: (any) => TUuid = generateDeterministicUUID)
```